### PR TITLE
Tuned PID Auto Balance

### PR DESCRIPTION
**Pull request by ericPrince**
@nathyelii I hope you don't mind me opening this pull request for you.

Summary:
- Tuned PD controller for auto-balance
- Auto-balance tuning parameters are displayed on the smart dashboard
- Lowered the auto drive backwards speed for testing (do we need to revert?)
- Decreased the sensed angle for triggering auto-balance (based on driving slower, and also the PID balance isn't as sensitive to the sensed angle, we just need to ensure it triggers)

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java
Comment (ericPrince) [inline src/main/java/frc/robot/RobotContainer.java:1]: @nathyelii @mikerada6 I think most of the changes to this file should be reverted. We added triggers for testing, but 1) the left xbox trigger needs to run the arm up, and 2) we found a way to run the command directly form the smart dashboard.

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -127,7 +132,7 @@ private void configureBindings() {
                 xButton.toggleOnTrue(new MoveArm(m_arm, ARM_DOWN, ARM_DOWN_SPEED).withTimeout(1.1));
                 leftBumper.whileTrue(new MoveWrist(m_wrist, WRIST_IN, MOTOR_SPEED));
                 rightBumper.whileTrue(new MoveWrist(m_wrist, WRIST_OUT, MOTOR_SPEED));
-                leftTigger.whileTrue(new MoveArm(m_arm, ARM_UP, ARM_UP_SPEED));
+                //leftTigger.whileTrue(new MoveArm(m_arm, ARM_UP, ARM_UP_SPEED));
```
Comment (mikerada6) [inline src/main/java/frc/robot/RobotContainer.java]: why did we remove this button.  does Gio only want to control the arm in one direction?

File: src/main/java/frc/robot/commands/AutoBalancePID.java [hunk 2]
```diff
@@ -13,17 +14,24 @@ public class AutoBalancePID  extends CommandBase {
...
 
     }
 
     @Override
     public void execute() {
-         double output = MathUtil.clamp(pid.calculate(m_drivetrain.getPitch(), 0), -0.4, 0.4);
-         m_drivetrain.arcadeDrive(-1*output,0);
+        double max_output = SmartDashboard.getNumber("Auto Balance Max Output", 0.53);
+        pid.setP(SmartDashboard.getNumber("Auto Balance P", 0.03));
+        pid.setD( SmartDashboard.getNumber("Auto Balance D", 0.004));
```
Comment (ericPrince) [inline src/main/java/frc/robot/commands/AutoBalancePID.java]: Don't forget to update these default values according to the final values we ended with today!

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -98,7 +100,10 @@ public RobotContainer() {
                                 m_drivetrain));
 
                 // Configure the trigger bindings
-                configureBindings();
+                configureBindings(); 
+
+                SmartDashboard.putData("Autonomous Balance PID",new AutoBalancePID(m_drivetrain));
```
Comment (ericPrince) [inline src/main/java/frc/robot/RobotContainer.java:92]: @mikerada6 here's how to add a command directly to the dashboard. Doing this causes a start button to pop up for the command, and while it's running there's a cancel button that you can push to stop it. It's really nice.

File: src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (ericPrince) [reply src/main/java/frc/robot/RobotContainer.java]: See my comment, this file should be reverted - that was for testing

**Pull request merged**